### PR TITLE
Support Quota datasource

### PR DESCRIPTION
**Pull request by P-Cao**
Added PowerScale Quota Datasource, AT tests and example

# Description

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE
- Feature Pull Request

##### RESOURCE OR DATASOURCE NAME

Quota Datasource

##### OUTPUT

image
image
image

##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Unit tests
- [x] Acceptance tests

### CODE REVIEW

**Comment by P-Cao** [comment]
run e2e test

**Comment by forrestxia** [comment]
run e2e test

**Comment by forrestxia** [comment]
@P-Cao please rebase your branch to pick up the fix from Connor

**Comment by P-Cao** [comment]
run e2e test

**Comment by P-Cao** [comment]
run e2e test

**Comment by P-Cao** [comment]
run e2e test

File: examples/data-sources/powerscale_quota/data-source.tf [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+# The SmartQuotas module is an optional quota-management tool that monitors and enforces administrator-defined storage limits.
+
+# Returns a list of PowerScale quotas based on filter block
+data "powerscale_quota" "example_quotas" {
+  filter {
```
Comment (taohe1012) [inline examples/data-sources/powerscale_quota/data-source.tf:22]: the quotas not support names ?

Comment (P-Cao) [reply examples/data-sources/powerscale_quota/data-source.tf:22]: No. Quota only has a poor readable id. And rest API already supports type/path filtering.

**Review by taohe1012 — APPROVED** [review]

**Review by forrestxia — APPROVED** [review]

**Comment by forrestxia** [comment]
run e2e test

**Comment by P-Cao** [comment]
run e2e test

**Comment by forrestxia** [comment]
run e2e test

**Comment by forrestxia** [comment]
e2e pass

**Comment by P-Cao** [comment]
run e2e test

**Comment by P-Cao** [comment]
image

**Comment by P-Cao** [comment]
e2e pass

**Comment by P-Cao** [comment]
run e2e test

**Comment by P-Cao** [comment]
run e2e test

**Pull request merged**